### PR TITLE
Switch to src-dir layout and move tests out-of-tree

### PR DESCRIPTION
**Pull request by bmerry**
This makes it more practical to run pytest against the installed code rather than the local tree.

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -7,7 +7,6 @@ include configure
 include include/spead2/common_loader_ibv.h
 include include/spead2/common_loader_mlx5dv.h
 include include/spead2/common_loader_rdmacm.h
-graft 3rdparty
```
Comment (bmerry) [inline MANIFEST.in:10]: Not really related, but I happened to notice this was still around even though that directory was removed a while ago.

File: setup.py [hunk 2]
```diff
@@ -170,7 +170,8 @@ def build_extension(self, ext):
         'pytest-timeout'
     ],
     python_requires='>=3.6',
-    packages=find_packages(),
+    packages=find_packages('src'),
+    package_dir={'': 'src'},
```
Comment (ludwigschwardt) [inline setup.py:174]: Recently went through this in my upcoming CALC package 😄

**Review by ludwigschwardt — APPROVED** [review]
Go `src`!

**Comment by bmerry** [comment]
Merged manually (by merging a downstream branch), so closing.

**Pull request closed without merging**